big.js

```diff
@@ -1,106 +1,302 @@
-window.onload = function() {
-  var s = document.querySelectorAll('body > div'), ti, i;
-  var notes = [];
-  for (i = 0; i < s.length; i++) {
-    s[i].setAttribute('tabindex', 0);
-    var noteElements = s[i].getElementsByTagName('notes');
-    notes.push([]);
-    while (noteElements.length) {
-      var note = noteElements[0];
-      notes[i].push(note.innerHTML.trim());
-      note.parentNode.removeChild(note);
-    }
+/* @flow weak */
+window.addEventListener('load', function() {
+  if (!document.body) {
+    throw new Error(
+      'big could not find a body element on this page, so it is exiting'
+    );
+  }
+
+  var body = document.body;
+  var slideDivs = nodeListToArray(document.querySelectorAll('body > div'));
+
+  if (!slideDivs.length) {
+    throw new Error(
+      "big couldn't find any slides in this presentation: " +
+        'there are no divs directly within the body of this page'
+    );
   }
-  if (!s.length) return;
-  var big = window.big = { current: 0, forward: fwd, reverse: rev, go: go, length: s.length,
-    audio: [].slice.call(document.getElementsByTagName('audio')).filter(function (audio) {
-      return audio.textTracks.length === 1 && audio.textTracks[0].cues.length > 0;
-    })[0] };
+
+  var ASPECT_RATIO = 1.6;
```

Comment:
I think [it's intentional](https://github.com/tmcw/big/pull/79#issue-222862055). seems like a good constraint for a project like big.